File: taller-5-python-ronalcastro97/zooAnimales/mamifero.py
from zooAnimales.animal import Animal

class Mamifero(Animal):
    _listado = []
    caballos = 0
    leones = 0
    
    def __init__(self, nombre, edad, habitat, genero, pelaje, patas):
        super().__init__(nombre, edad, habitat, genero)
        self._pelaje = pelaje
        self._patas = patas
        Mamifero._listado.append(self)

    @staticmethod
    def cantidadMamiferos():
        return len(Mamifero._listado)
    
    @classmethod
    def crearCaballo(cls, nombre, edad, genero):
        cls.caballos += 1 
        return cls(nombre, edad, "pradera", genero, True, 4)

    @classmethod
    def crearLeon(cls, nombre, edad, genero):
        cls.leones += 1 
        return cls(nombre, edad, "selva", genero, True, 4)

    def isPelaje(self):
        return self._pelaje

    def setPelaje(self, pelaje):
        self._pelaje = pelaje

    def getPatas(self):
        return self._patas
    
    def setPatas(self, patas):
        self._patas = patas

File: taller-5-java-ronalcastro97/src/main/java/zooAnimales/Mamifero.java
package zooAnimales;
import java.util.ArrayList;

public class Mamifero extends Animal{
	private  static ArrayList<Mamifero>listado=new ArrayList<>();
	public static  int caballos=0;
	public static int leones=0;
	private boolean pelaje;
	private  int patas;
	
	public Mamifero() {
		super();
		 Mamifero.listado.add(this);
	}
	
	public Mamifero(String nombre,int edad,String habitat,String genero,boolean pelaje,int patas){
		super(nombre,edad,habitat,genero);
		this.pelaje=pelaje;
		this.patas=patas;
		Mamifero.listado.add(this);
	}

	public static ArrayList<Mamifero> getListado() {
		return listado;
	}

	public  static void setListado(ArrayList<Mamifero> listado) {
		Mamifero.listado = listado;
	}

	public static int getCaballos() {
		return caballos;
	}

	public static void setCaballos(int caballos) {
		Mamifero.caballos = caballos;
	}

	public static int getLeones() {
		return leones;
	}

	public static void setLeones(int leones) {
		Mamifero.leones = leones;
	}

	public boolean isPelaje() {
		return pelaje;
	}

	public void setPelaje(boolean pelaje) {
		this.pelaje = pelaje;
	}

	public int getPatas() {
		return patas;
	}

	public void setPatas(int patas) {
		this.patas = patas; 
	} 
	public static Mamifero crearCaballo(String nombre,int edad,String genero) {
		
		Mamifero caballo= new Mamifero(nombre,edad,"Pradera",genero,true,4); 
		Mamifero.caballos++;
		return caballo;
		 
		
	} 
	public static Mamifero crearLeon(String nombre,int edad,String genero) {
		Mamifero leon= new Mamifero(nombre,edad,"selva",genero,true,4);
		Mamifero.leones++;
		return leon;
	}
	public static int cantidadMamiferos() {
		return Mamifero.listado.size();
	}
}
